src/app/ipv/middleware.js

```diff
@@ -303,10 +302,12 @@ module.exports = {
         );
       } else if (pageId === "pyi-triage-desktop-download-app") {
         // PYIC-4816: Use the actual device type selected on a previous page.
-        const qrCodeUrl =
-          SERVICE_URL + "/app-redirect/" + CONSTANTS.PHONE_TYPES.IPHONE;
+        const qrCodeUrl = SERVICE_URL + "/app-redirect/" + PHONE_TYPES.IPHONE;
         renderOptions.qrCode =
           await qrCodeHelper.generateQrCodeImageData(qrCodeUrl);
+      } else if (pageId === "pyi-triage-mobile-download-app") {
+        // PYIC-4816: Use the actual device type selected on a previous page and/or the current request's sniffed device type
+        renderOptions.appDownloadUrl = getAppStoreUrl(PHONE_TYPES.IPHONE);
```

Comment:
I'm starting to get a bit concerned about this function and how it handles so many different pages. If we stick with the idea that refreshing the `pyi-triage-desktop-download-app` page should do a check with core-back for the VC then that will probably need to be split out of here.